### PR TITLE
Improve download-latest.sh script: integrate apple-silicon binary

### PR DESCRIPTION
**Pull request by curquiza**
Fixes multiples issues: https://github.com/meilisearch/meilisearch/issues/3044, https://github.com/meilisearch/meilisearch/issues/3027, https://github.com/meilisearch/meilisearch/issues/2613, https://github.com/meilisearch/meilisearch/issues/3014

Improvement/Addition
- https://github.com/meilisearch/meilisearch/issues/3044
- https://github.com/meilisearch/meilisearch/issues/3027

Simplification
- With https://github.com/meilisearch/meilisearch/issues/3014, we removed the complex part to get the latest version of Meilisearch we can now get directly with the GitHub API

Bug fixes:
- Should remove the problem the users encountered in this issue (https://github.com/meilisearch/meilisearch/issues/2613) because the related part has been removed from the script

### CODE REVIEW

**Comment by curquiza** [comment]
@dureuill I put you as reviewer if you are curious since this is a change you spotted we had to do after v0.30.0 is released 😉

**Comment by curquiza** [comment]
For your information, I don't know why I cannot link the issues to the PR right now, looks like there is a bug on GitHub side, I will try later

File: download-latest.sh [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/bin/sh
 
+# This script works with a GitHub token to increase your request limit (for example, if using this script in a CI).
```
Comment (dureuill) [inline download-latest.sh:3]:
```suggestion
# This script can optionally use a GitHub token to increase your request limit (for example, if using this script in a CI).
```

Slight reword to make it clear that using a GitHub token, while useful, is optional (I had a doubt from the initial reading)

File: download-latest.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+# To make it work, fill the GITHUB_PAT environment variable with your GitHub token.
```
Comment (dureuill) [inline download-latest.sh:4]:
```suggestion
# To use a GitHub token, pass it through the GITHUB_PAT environment variable.
```

**Review by dureuill — COMMENTED** [review]
Thanks, this looks good to me! I just have some wording nitpicks, handle them as you wish.

I tested the script on Mac M1, didn't re-test on Windows/Linux/etc

**Review by dureuill — APPROVED** [review]

**Comment by curquiza** [comment]
@irevoire can you test on Linux? 😇

**Review by dureuill — APPROVED** [review]

**Comment by dureuill** [comment]
> @irevoire can you test on Linux? 😇

Hey @irevoire @curquiza,

Since I had a bit of time I tested under archlinux and was able to get a working Meilisearch 0.30.0 by using the script.

**Comment by curquiza** [comment]
Just discussed with @irevoire internally, he told me it's ok to merge

bors merge

**Pull request merged**